packages/server/src/sdk/app/applications/applications.ts

```diff
@@ -0,0 +1,56 @@
+import { AppStatus } from "../../../db/utils"
+import { App, ContextUser } from "@budibase/types"
+import { getLocksById } from "../../../utilities/redis"
+import { enrichApps } from "../../users/sessions"
+import { checkAppMetadata } from "../../../automations/logging"
+import { db as dbCore, users } from "@budibase/backend-core"
+
+export function filterAppList(user: ContextUser, apps: App[]) {
+  let appList: string[] = []
+  const roleApps = Object.keys(user.roles || {})
+  if (users.hasAppBuilderPermissions(user)) {
+    appList = user.builder?.apps || []
+    appList = appList.concat(roleApps)
+  } else if (!users.isAdminOrBuilder(user)) {
+    appList = roleApps
+  } else {
+    return apps
+  }
+  const finalApps: App[] = []
```

Comment:
This could be simplified in a `map`

Reply:
const finalApps = apps.filter(app => appList.includes(dbCore.getProdAppID(app.appId))